lib/macho/macho_file.rb

```diff
@@ -329,10 +329,13 @@ def get_mach_header
 
     # Read just the file's magic number and check its validity.
     # @return [Fixnum] the magic
+    # @raise [MachO::NotAMachOError] if the file is too small to have valid magic
     # @raise [MachO::MagicError] if the magic is not valid Mach-O magic
     # @raise [MachO::FatBinaryError] if the magic is for a Fat file
     # @private
     def get_and_check_magic
+      raise NotAMachOError.new("file is too small to be a Mach-O") if @raw_data.size < 4
```

Comment:
I think the limit should be 28 bytes (the size of `MachHeader`) instead of 4. I'd like to see a similar check in `FatFile#get_fat_header` with a limit of 8 bytes (the size of `FatHeader`; or 8+20+28 bytes if we consider a fat binary with no embedded Mach-O binaries to be invalid). See also my (yet to be written) non-code comment for more details.